src/Agent.Worker/TfManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    public static class TfUtil
+    {
+        public static async Task DownloadTfLegacyAsync(IExecutionContext executionContext)
+        {
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            string externalsPath = Path.Combine(executionContext.GetVariableValueOrDefault("Agent.HomeDirectory"), Constants.Path.ExternalsDirectory);
+            ArgUtil.NotNull(externalsPath, nameof(externalsPath));
+
+            string tfLegacyExternalsPath = Path.Combine(externalsPath, "tf-legacy");
+
+            if (!Directory.Exists(tfLegacyExternalsPath))
```

Comment:
Both download sections looks similar. The only difference is the package name and destination path.
Can you consider a refactoring of this?

Reply:
Fixed